modules/virtualization/microvm/audiovm.nix

```diff
@@ -0,0 +1,116 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  ...
+}: let
+  configHost = config;
+  vmName = "audio-vm";
+  macAddress = "02:00:00:02:03:04";
+  audiovmBaseConfiguration = {
+    imports = [
+      (import ./common/vm-networking.nix {
+        inherit vmName;
+        inherit macAddress;
+      })
+      ({
+        lib,
+        pkgs,
+        ...
+      }: {
+        ghaf = {
+          users.accounts.enable = lib.mkDefault configHost.ghaf.users.accounts.enable;
+          profiles.graphics.enable = false;
+          development = {
+            ssh.daemon.enable = lib.mkDefault configHost.ghaf.development.ssh.daemon.enable;
+            debug.tools.enable = lib.mkDefault configHost.ghaf.development.debug.tools.enable;
+          };
+        };
+
+        environment = {
+          systemPackages = [
+            pkgs.pamixer
+          ];
+        };
+
+        system.stateVersion = lib.trivial.release;
+
+        nixpkgs.buildPlatform.system = configHost.nixpkgs.buildPlatform.system;
+        nixpkgs.hostPlatform.system = configHost.nixpkgs.hostPlatform.system;
+
+        time.timeZone = "Asia/Dubai";
+
+        # Enable pulseaudio support for host as a service
+        sound.enable = true;
+        hardware.pulseaudio.enable = true;
+        hardware.pulseaudio.systemWide = true;
+        networking.firewall.allowedTCPPorts = [4713];
+
+        # Allow microvm user to access pulseaudio
+        users.extraUsers.ghaf.extraGroups = ["audio" "pulse-access"];
+        # Enable and allow TCP connection from localhost and host only
+        hardware.pulseaudio.tcp.enable = true;
+        hardware.pulseaudio.tcp.anonymousClients.allowedIpRanges = ["127.0.0.1" "192.168.101.2"];
```

Comment:
Both `127.0.0.1` and `192.168.101.2` could be defined as variables under `let`?

Reply:
I agree that they could. But i also think that the definition should be done somewhere globally rather than in audiovm.nix